core-bundle/src/Resources/contao/controllers/BackendPassword.php

```diff
@@ -127,6 +133,46 @@ public function run()
 
 		return $objTemplate->getResponse();
 	}
+
+	/**
+	 * Evaluate custom regex. See https://github.com/contao/contao/issues/5316
+	 *
+	 * @param  string      $pw The password
+	 * @return bool|string The error message, false otherwise
+	 */
+	private function _evalRegexp($pw)
+	{
+		// no custom regex available
+		if (!isset($GLOBALS['TL_HOOKS']['addCustomRegexp']) || !\is_array($GLOBALS['TL_HOOKS']['addCustomRegexp']))
+		{
+			return false;
+		}
+
+		// load user data container
+		$this->loadDataContainer('tl_user');
+		$dc = new DC_Table('tl_user');
+		$dc->id = $this->User->id;
+
+		// load regex
+		$rgxp = $GLOBALS['TL_DCA']['tl_user']['fields']['password']['eval']['rgxp'];
+		$widget = new Password(Password::getAttributesFromDca($GLOBALS['TL_DCA']['tl_user']['fields']['password'], 'password'));
+		$widget->dataContainer = $dc;
+
+		// iterate over available regex hooks
+		foreach ($GLOBALS['TL_HOOKS']['addCustomRegexp'] as $callback)
+		{
+			$this->import($callback[0]);
+			$break = $this->{$callback[0]}->{$callback[1]}($rgxp, $pw, $widget);
+
+			// return the actual error message if it exists
+			if ($break === true && $widget->hasErrors())
+			{
+				return $widget->getErrorsAsString();
+			}
+		}
```

Comment:
You don't need to call the hook yourself, I think you can just execute `$widget->validate()`.